### PR TITLE
Prevent a namespace from being deleted if it is the target of a Nexus endpoint

### PR DESCRIPTION
**Pull request by bergundy**
## Why?

Avoid breaking callers.

## How did you test it?

Added unit test assertions.

### CODE REVIEW

File: service/frontend/operator_handler.go [hunk 1]
```diff
@@ -611,6 +614,38 @@ func (h *OperatorHandlerImpl) DeleteNamespace(
...
+		})
+		if err != nil {
+			return nil, err
+		}
+		for _, entry := range resp.GetEndpoints() {
+			if nsName := entry.GetSpec().GetTarget().GetWorker().GetNamespace(); nsName == requestNSName {
+				return nil, serviceerror.NewFailedPrecondition(fmt.Sprintf("cannot delete a namespace that is a target of a Nexus endpoint (%s)", entry.GetSpec().GetName()))
+			}
+		}
+		if len(nextPageToken) == 0 {
```
Comment (pdoerner) [inline service/frontend/operator_handler.go:644]: You're not updating `nextPageToken` inside the loop.

File: service/frontend/operator_handler.go [hunk 2]
```diff
@@ -611,6 +614,38 @@ func (h *OperatorHandlerImpl) DeleteNamespace(
...
+	requestNSName := request.GetNamespace()
+	if request.GetNamespaceId() != "" {
+		nsName, err := h.namespaceRegistry.GetNamespaceName(namespace.ID(request.GetNamespaceId()))
+		if err != nil {
+			return nil, err
+		}
+		requestNSName = nsName.String()
+	}
+
+	// Prevent deletion of a namespace that is associated with any Nexus endpoints.
```
Comment (pdoerner) [inline service/frontend/operator_handler.go:630]: nit:

```suggestion
	// Prevent deletion of a namespace that is the target of any Nexus endpoints.
```

**Review by pdoerner — COMMENTED** [review]
Might want to also add a test that covers paginating over the endpoints list.

File: service/frontend/operator_handler.go (hunk 1 above)
Comment (bergundy) [reply service/frontend/operator_handler.go:644]: 🤦 thanks!

File: common/dynamicconfig/constants.go [hunk 3]
```diff
@@ -865,6 +865,13 @@ and deployment interaction in matching and history.`,
 		`EnableNexus toggles all Nexus functionality on the server. Note that toggling this requires restarting
 server hosts for it to take effect.`,
 	)
+
+	AllowDeleteNamespaceIfNexusEndpointTarget = NewGlobalBoolSetting(
+		"frontend.allowDeleteNamespaceIfNexusEndpointTarget",
+		false,
+		`If set to true (default) namespaces that are Nexus endpoint targets will be prevented from being deleted.`,
```
Comment (rodrigozhou) [inline common/dynamicconfig/constants.go:872]: Default value is not `true`.

**Review by rodrigozhou — APPROVED** [review]

**Review by pdoerner — APPROVED** [review]

File: common/dynamicconfig/constants.go (hunk 3 above)
Comment (bergundy) [inline common/dynamicconfig/constants.go:872]:
```suggestion
		`If set to true (default is false), namespaces that are Nexus endpoint targets will be prevented from being deleted.`,
```

**Pull request merged**